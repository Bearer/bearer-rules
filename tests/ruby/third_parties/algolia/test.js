const { createInvoker, getEnvironment } = require("../../../helper.js")
const { ruleId, ruleFile, testBase } = getEnvironment(__dirname)

describe(ruleId, () => {
  const invoke = createInvoker(ruleId, ruleFile, testBase)
  
  test("datatype_in_save_object", ({ expect }) => {
    const testCase = "datatype_in_save_object.rb"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
  test("ok_datatype_ignored", ({ expect }) => {
    const testCase = "ok_datatype_ignored.rb"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
  test("ok_no_datatype", ({ expect }) => {
    const testCase = "ok_no_datatype.rb"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
})