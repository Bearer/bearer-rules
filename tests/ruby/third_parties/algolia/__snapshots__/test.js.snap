// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ruby_third_parties_algolia datatype_in_save_object 1`] = `
"{
  "high": [
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_algolia",
      "title": "Leakage of sensitive data to Algolia",
      "description": "## Description\\nLeaking sensitive data to third-party data tools is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Algolia.\\n\\n## Remediations\\nWhen sending data to third-party services, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Algolia docs](https://www.algolia.com/doc/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_algolia",
      "line_number": 4,
      "full_filename": "/tmp/bearer-scan/datatype_in_save_object.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 4,
        "end": 4,
        "column": {
          "start": 28,
          "end": 38
        }
      },
      "sink": {
        "start": 4,
        "end": 4,
        "column": {
          "start": 1,
          "end": 89
        },
        "content": "index.save_object({ email: user.email }, { auto_generate_object_id_if_not_exist: true })"
      },
      "parent_line_number": 4,
      "snippet": "index.save_object({ email: user.email }, { auto_generate_object_id_if_not_exist: true })",
      "fingerprint": "831b70f679c5e1d92b8f4d4f973f7dc4_0",
      "old_fingerprint": "b4d725160def8af2226e01c3057c9bc6_0",
      "code_extract": "index.save_object({ email: user.email }, { auto_generate_object_id_if_not_exist: true })"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_algolia",
      "title": "Leakage of sensitive data to Algolia",
      "description": "## Description\\nLeaking sensitive data to third-party data tools is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Algolia.\\n\\n## Remediations\\nWhen sending data to third-party services, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Algolia docs](https://www.algolia.com/doc/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_algolia",
      "line_number": 6,
      "full_filename": "/tmp/bearer-scan/datatype_in_save_object.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 6,
        "end": 6,
        "column": {
          "start": 30,
          "end": 40
        }
      },
      "sink": {
        "start": 6,
        "end": 6,
        "column": {
          "start": 1,
          "end": 92
        },
        "content": "index.save_objects([{ email: user.email }], { auto_generate_object_id_if_not_exist: true })"
      },
      "parent_line_number": 6,
      "snippet": "index.save_objects([{ email: user.email }], { auto_generate_object_id_if_not_exist: true })",
      "fingerprint": "831b70f679c5e1d92b8f4d4f973f7dc4_1",
      "old_fingerprint": "b4d725160def8af2226e01c3057c9bc6_1",
      "code_extract": "index.save_objects([{ email: user.email }], { auto_generate_object_id_if_not_exist: true })"
    }
  ]
}"
`;

exports[`ruby_third_parties_algolia ok_datatype_ignored 1`] = `"{}"`;

exports[`ruby_third_parties_algolia ok_no_datatype 1`] = `"{}"`;
